### DISCUSSION TITLE
Issue with theme not being respected on certain elements

### DISCUSSION THREAD

**Discussion by greg-bishop** [microsoft/fluentui-blazor#2593] [Q&A]
I am building a custom Teams app.  Naturally, I'm using the Teams color scheme, and I have the ability for the user to change the mode (System, Light, and Dark).

I have the following in App.razor:
```
<FluentDesignTheme OfficeColor="OfficeColor.Teams" StorageName="theme" />
```

And I have the following in a component that allows the user to change the mode:
```
<FluentDesignTheme @bind-Mode="Mode" StorageName="theme" OnLuminanceChanged="OnLuminanceChanged" />
```

For most elements, the color and mode are being respected properly.  However, on FluentSwitches, it is still using Light mode (note the text color below) with a blue color scheme:
![image](https://github.com/user-attachments/assets/a2b01b89-160b-49d9-898a-8f18c51b373c)

The FluentMenu is also using Light mode and a blue color scheme:
![image](https://github.com/user-attachments/assets/a6c9a7a1-e2c7-418a-a7a2-9709414d6221)

As expected, the styles from the #shadow-root :host are overriding the styles from :root, but the issue is that the :host styles are not respecting the theme even though the :root styles are.

I have tried removing the FluentDesignTheme that allows the user to change the mode and hardcoding both Mode and OfficeColor in the FluentDesignTheme in App.razor, but the issue remained.

Has anyone else run into this?  Is there something I can change to fix this?

_System Info:_
Browser: Google Chrome v128.0.6613.85
Microsoft.FluentUI.AspNetCore.Components v4.9.3

**Comment by dvoituron** [comment]
The switch color is defined by this CSS and an example is on the Demo website:
![{AB8CA9AB-5839-4ED9-96E2-A42399E1D213}](https://github.com/user-attachments/assets/25f27208-321d-47de-8ef8-0667bcd93bd6)


```css
:host(.checked) .switch {
    background: var(--accent-fill-rest);
}
```

Can you check that you have this style, and that it's not overloaded?

And that is similar for the FluentMenu in Dark mode.

![{0B034CAA-571D-421C-931E-A678C8B5FB24}](https://github.com/user-attachments/assets/af0d1b37-1aaa-4fe9-abd5-cdd1e0e7b12c)

**Reply by greg-bishop** [reply]
Thank you for your response.  I do have this style, and it is using the variable.

![image](https://github.com/user-attachments/assets/14d88888-1f46-4484-816d-b7a44e5eb02e)

The issue I'm running into is that the :host variables are not respecting the theme settings, even though the :root variables are.

![image](https://github.com/user-attachments/assets/fedf57e7-6e73-44f2-97a4-2989feb4f501)

Here's some of the :root variables:

![image](https://github.com/user-attachments/assets/a6b29f7f-0e3f-4633-bd34-10c033b0be1b)

In the demo website, the :host variables change when the theme changes, so even though the :host variables override the :root variables, the theme is consistently applied:

![image](https://github.com/user-attachments/assets/dc4dc097-6bf5-4e91-984f-05e200c9201d)
